image-build/capg: add periodic job to build gce images

cpanato opened the pull request:
- remove the temporary job created
- add a periodic job to build the gce images

/hold for https://github.com/kubernetes-sigs/image-builder/pull/445

/assign @CecileRobertMichon @dims @codenrhoden


TODO:
- create a GCP project to have the images published
- change this periodic to use the project and publish nightly builds

dims commented on config/jobs/kubernetes-sigs/image-builder/image-builder-periodics.yaml:
```diff
@@ -0,0 +1,31 @@
+          - "./images/capi/scripts/ci-gce.sh"
+        env:
+          - name: BOSKOS_HOST
+            value: "boskos.test-pods.svc.cluster.local"
```
Using boskos will mean that the image will be uploaded to random projects that boskos picks up. So no, let's not do this please. we want to push the image specifically to a project (like `"k8s-staging-cloud-provider-gcp"`)

cpanato replied:
ok, thanks for the feedback.
I was thinking a bit more and I will create another script for the nightly and will build images for 1.18/1.19/1.20/1.21 the default case only builds for 1.18.

and one question: when a new job run to we want to delete the image created a day before? or overwrite that (I don't think is possible to overwrite, but then delete and recreate them)

cpanato replied:
made the changes here and in the PR https://github.com/kubernetes-sigs/image-builder/pull/445 to have a dedicated script

dims replied:
thanks!

dims commented:
/retitle [WIP] image-build/capg: add periodic job to build gce images

dims commented on config/jobs/image-pushing/k8s-staging-cluster-api.yaml:
```diff
@@ -258,60 +258,3 @@ periodics:
     testgrid-alert-email: sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io
-
-##
-## This Job is a temporary solution that builds the machine images using image-builder repository
```
please dont delete this yet @cpanato

dims commented:
@cpanato needs a rebase

dims commented:
@cpanato there are still 2 issues we will have to work through:

- Does the service account that gets injected have access to the CAPG staging cluster
- Does the CAPG staging cluster have gcloud compute api service enabled.

codenrhoden commented:
I honestly didn't know that image-pushing config was even there for GCP (not an area that I've ever been involved with). But what's happening here makes sense to me.

cpanato commented:
Added the same owners as described https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/image-builder/OWNERS
Don't know if need to add the CAPG maintainers as well 


added the periodic job inside the wg-infra trusted, see the PR for the enable the API and associations for the trusted cluster: https://github.com/kubernetes/k8s.io/pull/2025

cc @dims @ameukam @spiffxp

ameukam commented:
@cpanato I don't particularly mind having this job here but I think https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/image-builder is a better place for this job.

cpanato commented:
> @cpanato I don't particularly mind having this job here but I think https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/image-builder is a better place for this job.

But to run in the trusted cluster i need to add the job in this directory right? Only if dont need to run in the trused cluster then i can rearrange and update the service account attribution.

Did that because Aaron post in the PR the trusted thing

cpanato commented:
/hold cancel

https://github.com/kubernetes/k8s.io/pull/2025 is merged and we can merge https://github.com/kubernetes-sigs/image-builder/pull/445

so I think we are good to move on for the validation part
@ameukam @spiffxp

ameukam commented:
/approve
/lgtm
(not sure if I have the permission)

ameukam commented:
> > @cpanato I don't particularly mind having this job here but I think https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/image-builder is a better place for this job.
> 
> But to run in the trusted cluster i need to add the job in this directory right? Only if dont need to run in the trused cluster then i can rearrange and update the service account attribution.

You don't necessarily need to add a job in this specific to make it run in `k8s-infra-prow-build-trusted`. For  instance `post-cluster-api-push-images` is in a different directory: https://github.com/kubernetes/test-infra/blob/c7b24a4c02ca978040dc802b3b37a14fd6f69d61/config/jobs/image-pushing/k8s-staging-cluster-api.yaml#L3-L6

cpanato commented:
@ameukam ahh, will open a PR to move this out
